provider/lib.go

```diff
@@ -142,3 +142,30 @@ func MultilineGrep(ctx context.Context, window int, path, pattern string) (int,
 
 	return -1, scanner.Err()
 }
+
+// GetIncludedPathsFromConfig returns validated includedPaths from provider settings
+// if allowFilePaths is not set, path to a file is converted into a path to its base dir
+func GetIncludedPathsFromConfig(i InitConfig, allowFilePaths bool) []string {
+	validatedPaths := []string{}
+	if includedPaths, ok := i.ProviderSpecificConfig[IncludedPathsConfigKey].([]interface{}); ok {
```

Comment:
Do we want to error, or warn if it is not a slice?

Just to make debug of provider config easier?

Reply:
@shawn-hurley good point.